internals/models.py

```diff
@@ -955,38 +955,62 @@ def get_in_milestone(
     if milestone == None:
       return None
 
+    all_features = {}
+    all_features[IMPLEMENTATION_STATUS[ENABLED_BY_DEFAULT]] = []
+    all_features[IMPLEMENTATION_STATUS[DEPRECATED]] = []
+    all_features[IMPLEMENTATION_STATUS[REMOVED]] = []
+    all_features[IMPLEMENTATION_STATUS[INTERVENTION]] = []
+    all_features[IMPLEMENTATION_STATUS[ORIGIN_TRIAL]] = [] 
+    all_features[IMPLEMENTATION_STATUS[BEHIND_A_FLAG]] = []
+    
     logging.info('Getting chronological feature list in milestone %d', milestone)
     q = Feature.query()
     q = q.order(Feature.name)
     q = q.filter(Feature.shipped_milestone == milestone)
     desktop_shipping_features = q.fetch(None)
+    
+    # Push feature to list corresponding to the implementation status of feature in queried milestone
     for feature in desktop_shipping_features:
-      if feature.impl_status_chrome not in {ENABLED_BY_DEFAULT, DEPRECATED, REMOVED}:
-        if feature.feature_type == FEATURE_TYPE_DEPRECATION_ID and Feature.dt_milestone_desktop_start != None:
-          feature.impl_status_chrome = DEPRECATED
-        if feature.feature_type == FEATURE_TYPE_INCUBATE_ID:
-          feature.impl_status_chrome = ENABLED_BY_DEFAULT
+      if feature.impl_status_chrome == ENABLED_BY_DEFAULT:
+        all_features[IMPLEMENTATION_STATUS[ENABLED_BY_DEFAULT]].append(feature)
```

Comment:
I was thinking that the features would be grouped by the milestone that matched the requested milestone, not by the current status of the feature.  However, let's go ahead with what you have here.  We can always come back to it.